src/Avalonia.Controls/Primitives/ScrollBar.cs

```diff
@@ -193,6 +250,68 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
             }
         }
 
+        private void InvokeAfterDelay(Action handler, TimeSpan delay)
```

Comment:
interesting, i had thought we could use transition delay, etc.

Reply:
Transitions have several issues right now that make use hard:
- you can't bind any properties on a transition.
- in this case we have different delay when collapsing and when expanding. CSS way of solving that is to have two sets of transitions, one that matches non-expanded and one that matches the expanded one. But due to bugs in transition handling replacing transitions cause neutral values to be lost.